### ISSUE TITLE
Docker file uploader generates unknown error with "pty": true

### ISSUE THREAD

**Issue by avishefi** [mitchellh/packer#2581]
Trying to upload a folder to a Docker builder using the following configuration (using `"pty": true` in builder config) generates an error during the upload step. Using `packer -debug` and `PACKER_LOG=1` doesn't generate additional information that allows interrogating what's the real error.
#### With `"pty": true`

```
2015/08/11 16:44:13 packer-builder-docker: 2015/08/11 16:44:13 Executing in container eda3e9302bd9f7ce7db2fb49338954f5d264d0c9223a78ea0be5f22d6d0ccd56: "(set -e; mkdir -p /user/installer; command cp -R /packer-files/dirupload278569253/* /user/installer) >/packer-files/cmd578545728 2>&1; echo $? >/packer-files/cmd578545728-exit"
2015/08/11 16:44:15 ui: ==> docker: Killing the container: eda3e9302bd9f7ce7db2fb49338954f5d264d0c9223a78ea0be5f22d6d0ccd56
2015/08/11 16:44:16 ui error: Build 'docker' errored: Upload failed with non-zero exit status: 
2015/08/11 16:44:16 Builds completed. Waiting on interrupt barrier...
2015/08/11 16:44:16 machine readable: error-count []string{"1"}
2015/08/11 16:44:16 ui error: 
==> Some builds didn't complete successfully and had errors:
2015/08/11 16:44:16 machine readable: docker,error []string{"Upload failed with non-zero exit status: 1"}
2015/08/11 16:44:16 ui error: --> docker: Upload failed with non-zero exit status: 1
2015/08/11 16:44:16 ui: 
```
#### Without `"pty": true`

```
2015/08/11 16:41:37 packer-builder-docker: 2015/08/11 16:41:37 Executing in container 417994d683f9f9730adb0813a8a287df0a0ae9423270c59f3e33150469ee844a: "(set -e; mkdir -p /user/installer; command cp -R /packer-files/dirupload183934075/* /user/installer) >/packer-files/cmd639900062 2>&1; echo $? >/packer-files/cmd639900062-exit"
2015/08/11 16:41:39 packer-builder-docker: 2015/08/11 16:41:39 Waiting for exit code to appear for remote command...
2015/08/11 16:41:39 packer-builder-docker: 2015/08/11 16:41:39 Executed command exit status: 0
```
#### Packer template

``` json
{
    "builders": [{
        "type": "docker",
        "image": "...",
        "commit": true,
        "pty": true
    }],
    "provisioners": [
        {
            "type": "file",
            "source": "installer",
            "destination": "/user"
        },
        {
            "type": "shell",
            "script": "provision.sh"
        }
    ]
}
```

**Comment by cbednarski** [comment]
Thanks for the report! I have not looked into this yet but I have a funny feeling we may need to suppress the pty option for this operation. I am planning to look at this area of the code today so I will check it out while I'm there.

**Comment by iroller** [comment]
Same issue here.

**Comment by mwhooker** [comment]
I think this will be resolved by #4186. After that gets merged, please open a new issue if you continue to see this
